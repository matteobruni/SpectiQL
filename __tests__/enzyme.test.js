import App from "../client/App.jsx";
import GenerateTest from "../client/Components/GenerateTest.jsx";
import TestInput from "../client/Components/TestInput.jsx";
import TestQuery from "../client/Components/TestQuery.jsx";
import MutationTestInput from "../client/Components/MutationTestInput.jsx";
import LeftSideBar from "../client/Components/LeftSideBar.jsx";
import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import toJson from "enzyme-to-json";

//enzyme configuration
configure({ adapter: new Adapter() });

//testing App component
describe("testing App component", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<App />);
  });
  it("renders without crashing", () => {
    expect(wrapper.length).toEqual(1);
  });
});

// testing Generate Test for Queries
describe("testing Generate Test component", () => {
  let wrapper;
  const props = {
    testDescription: "test description",
    testSuiteName: "test suite name",
    value: "random value"
  };
  beforeAll(() => {
    wrapper = shallow(<GenerateTest {...props} />);
  });
  it("Snapshot testing", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("contains a form with 2 buttons", () => {
    expect(wrapper.find("Button")).toHaveLength(2);
  });
});

// testing for TestInput
describe("testing Test Input component", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<TestInput />);
  });

  it("snapshot testing", () => {
    expect(wrapper).toMatchSnapshot();
  });
  it("has 7 options within dropdown", () => {
    expect(wrapper.find("option")).toHaveLength(7);
  });
});

// test for TestQuery
describe("testing TestQuery component", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<TestQuery />);
  });

  it("Snapshot testing", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it("should render a form for write query", () => {
    expect(wrapper.find("div.writeQueryBox").exists()).toBe(true);
  });
});

// test for MutationTestInput
describe("testing MutationTestInput component", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<MutationTestInput />);
  });
  it("should render a form for write query", () => {
    expect(wrapper.find("div.testInputContainer").exists()).toBe(true);
  });
});

//test for leftSideBar
describe("testing for leftSideBar", () => {
  let wrapper;
  beforeAll(() => {
    wrapper = shallow(<LeftSideBar />);
  });
  it("Snapshot testing", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  it("should render a button toolbar with 4 buttons", () => {
    expect(wrapper.find("div.mainNavBar").exists()).toBe(true);
    expect(wrapper.find("Button")).toHaveLength(4);
  });
});
