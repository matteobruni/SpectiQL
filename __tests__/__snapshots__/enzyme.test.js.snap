// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Generate Test component Snapshot testing 1`] = `
<div
  className="generateTestBox"
>
  <FormGroup
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Col",
        "render": [Function],
      }
    }
    style={
      Object {
        "margin": 0,
        "padding": 0,
      }
    }
  >
    <Button
      active={false}
      block={true}
      className="generateTestButton"
      disabled={false}
      size="lg"
      style={
        Object {
          "borderColor": "black",
          "color": "black",
          "height": "49%",
          "marginBottom": "1.5%",
          "width": "100%",
        }
      }
      type="button"
      variant="outline-primary"
    >
      Generate Test
    </Button>
    <Button
      active={false}
      block={true}
      className="addTestButton"
      disabled={false}
      size="lg"
      style={
        Object {
          "borderColor": "black",
          "color": "black",
          "height": "49.7%",
          "width": "100%",
        }
      }
      type="button"
      variant="outline-primary"
    >
      Update Test Suite
    </Button>
  </FormGroup>
  <FormGroup
    as={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "displayName": "Col",
        "render": [Function],
      }
    }
    className="generatedTestForm"
    controlId="sample-test"
  >
    <FormControl
      as="textarea"
      className="generatedTestForm"
      placeholder="Generated Test..."
      style={
        Object {
          "height": "100%",
          "width": "100%",
        }
      }
    />
  </FormGroup>
</div>
`;

exports[`testing Test Input component snapshot testing 1`] = `ShallowWrapper {}`;

exports[`testing TestQuery component Snapshot testing 1`] = `
<div
  className="writeQueryBox"
>
  <div
    className="test-query-write"
  >
    <FormGroup
      as={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "displayName": "Col",
          "render": [Function],
        }
      }
      className="writeQueryForm"
      controlId="write-query"
    >
      <FormControl
        as="textarea"
        name="writeTest"
        placeholder="Enter query here..."
        style={
          Object {
            "height": "98%",
            "marginBottom": "2%",
            "width": "100%",
          }
        }
      />
    </FormGroup>
  </div>
</div>
`;

exports[`testing for leftSideBar Snapshot testing 1`] = `
<div
  className="mainNavBar"
>
  <ButtonToolbar
    className="buttonContainer"
    role="toolbar"
  >
    <Link
      className="queries"
      to="/queries"
    >
      <Button
        active={false}
        className="queries"
        disabled={false}
        size="lg"
        type="button"
        variant="outline-dark"
      >
        Queries
      </Button>
    </Link>
    <Link
      className="mutations"
      to="/mutations"
    >
      <Button
        active={false}
        className="mutations"
        disabled={false}
        size="lg"
        type="button"
        variant="outline-dark"
      >
        Mutations
      </Button>
    </Link>
    <Button
      active={false}
      className="subscriptions"
      disabled={false}
      size="lg"
      type="button"
      variant="outline-dark"
    >
      Subscriptions
    </Button>
    <Button
      active={false}
      className="export"
      disabled={false}
      size="lg"
      type="button"
      variant="outline-dark"
    >
      Export
    </Button>
  </ButtonToolbar>
</div>
`;
